src/MX/MegaMenu/Model/Menu/Item.php

```diff
@@ -191,11 +197,16 @@ protected function getCategories($item)
      */
     protected function getCategory($categoryIdPath)
     {
-        $contentCategory = explode('/', $categoryIdPath);
-        if ($contentCategory && isset($contentCategory[self::OFFSET_CATEGORY_ID])) {
-            $categoryId = $contentCategory[self::OFFSET_CATEGORY_ID];
+        try {
+            $contentCategory = explode('/', $categoryIdPath);
+            if ($contentCategory && isset($contentCategory[self::OFFSET_CATEGORY_ID])) {
+                $categoryId = $contentCategory[self::OFFSET_CATEGORY_ID];
 
-            return $this->categoryRepository->get($categoryId);
+                return $this->categoryRepository->get($categoryId);
+            }
+        } catch (NoSuchEntityException $error) {
+            // Magento Category Repository throws NoSuchEntityException when no category found. That shouldn't break the FE rendering
+            $this->_logger->addError($error->getMessage());
```

Comment:
we should use the `Psr\Log\LoggerInterface` methods only on `_logger`, and I would always log the whole exception because it can then add a stack trace, e.g. `$this->_logger->error($error);`

Reply:
Actually the Psr\Log\LoggerInterface is used and it's available via $this->_logger in the AbstractModel. I'll amend the stack trace displaying.